lib/galaxy/datatypes/tabular.py

```diff
@@ -154,12 +154,12 @@ def get_chunk(self, trans, dataset: HasFileName, offset: int = 0, ck_size: Optio
         )
 
     def _read_chunk(self, trans, dataset: HasFileName, offset: int, ck_size: Optional[int] = None):
-        with compression_utils.get_fileobj(dataset.get_file_name()) as f:
+        with compression_utils.get_fileobj(dataset.get_file_name(), "rb") as f:
             f.seek(offset)
             ck_data = f.read(ck_size or trans.app.config.display_chunk_size)
-            if ck_data and ck_data[-1] != "\n":
+            if ck_data and ck_data[-1] != b"\n":
                 cursor = f.read(1)
-                while cursor and cursor != "\n":
+                while cursor and cursor != b"\n":
```

Comment:
Reading single chars until you hit a newline was never ideal, but with a binary file this could potentially take a really long time, couldn't it?

Reply:
arg, yes, we're not reading from from the chunk ...